riscemu/registers.py

```diff
@@ -156,7 +159,13 @@ def _reg_repr(self, reg: str, name_len=4, fmt="08X"):
             return FMT_ORANGE + FMT_UNDERLINE + txt + FMT_NONE
         if reg == "zero":
             return txt
-        if self.get(reg, False) == 0 and reg not in Registers.named_registers():
+        if (
+            reg in self.valid_regs
+            and self.get(reg, False) == 0
+            and reg not in Registers.named_registers()
+            or reg in self.float_regs
+            and self.get_f(reg, False)
+        ):
```

Comment:
What in tarnation? Can we make this a bit more readable?

Reply:
> What in tarnation? Can we make this a bit more readable?

I had to create a seperate formatting condition for changing empty float registers into the gray colour so that it also uses get_f instead of get. It started out as two seperate if statements, one for regular registers (reg in self.valid_regs and self.get(reg, False) == 0 and reg not in Registers.named_registers()) and one for float registers (reg in self.float_regs and self.get_f(reg, False)), but I decided to combine them with an or since they led to the same result.

Reply:
Maybe you can also do something like this?

```py
should_color_float = reg in self.float_regs and self.get_f(reg, False) != 0
should_color_int = reg in self.valid_regs and (
    self.get(reg, False) != 0 
    or reg in Registers.named_registers()
)

if should_color_float or should_color_int:
    # print colored
else:
    # print grey
```

Reply:
ye, sounds good as well, I'm on it